examples/github-repositories-custom-plugin/createGitHubReposPlugin.tsx

```diff
@@ -0,0 +1,127 @@
+import { AutocompletePlugin } from '@algolia/autocomplete-js';
+import qs from 'qs';
+import unfetch from 'unfetch';
+import debounce from 'debounce-promise';
+
+type GitHubRepository = {
+  full_name: string,
+  description: string,
+  stargazers_count: number,
+  html_url: string,
+  owner: {
+    avatar_url: string,
+  },
+};
+
+type CreateGithubReposPluginProps = {
+  accept?: string,
+  sort?: 'stars' | 'forks' | 'help-wanted-issues' | 'updated',
+  order?: 'asc' | 'desc',
+  per_page?: number,
+  page?: number,
+};
+
+const debouncedFetch = debounce(unfetch, 300);
```

Comment:
Yes! I took some time to reimplement a `debouncePromise` function myself in the demo, so we could drop the `unfetch` (not really needed) and `debounce-fetch` (unmaintained) dependencies from the code (see 1b7606441d57cc0831b8b41319407d997383a197).

I think debouncing at the promise level is an interesting pattern because it makes the user code pretty much similar to not debouncing. It would also work with `getAlgoliaHits` or whatever fetcher that returns a promise.

But then, what do you think of providing it at the API level? Autocomplete v0 did have a `debounce` option, which is nice to have in the context of search. Do you think it's worth extending the API surface for it, or do we want to keep this in userland?

Reply:
For now as long as it's possible user-land it's good. I reckon we'll need to support that in the library once we start introducing Algolia Answers helpers, where debouncing is required.

Autocomplete v0 allows debouncing for all sources at once, but in this new version, we might rather want to debounce per source (that'll be part of our Sequencer API).